File: JAVA PROGRAMMING/Test51.java
// 51. PRINT PERFECT SQUARE NUMBERS FROM RANGE1 TO RANGE2
import java.util.Scanner;
public class Test51 {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter Range1 Value: ");
        int range1 = in.nextInt();

        System.out.println("Enter Range2 Value: ");
        int range2 = in.nextInt();

        int start = (int) Math.ceil(Math.sqrt(range1));

        for(int i = start; i * i <= range2; i++) {
            System.out.print(i * i+" ");
        }
        in.close();
    }
}

File: TYPESCRIPT PROGRAMMING/Test51.ts
// 51. PRINT PERFECT SQUARE NUMBERS FROM RANGE1 TO RANGE2
const PerfectSqaureRange = (range1: number, range2: number): void => {
    let start: number = Math.ceil(Math.sqrt(range1));
    for(let i: number = start; i * i <= range2; i++) {
        process.stdout.write(`${i * i} `);
    }
    console.log();
}
PerfectSqaureRange(100, 200);
PerfectSqaureRange(100, 500);
PerfectSqaureRange(500, 1000);